test/distributed/pipelining/test_schedule.py

```diff
@@ -344,7 +362,174 @@ def test_grad_with_manual_interleaved(self, ScheduleClass):
 
 instantiate_parametrized_tests(ScheduleTest)
 
+
+def format_pipeline_order(pipeline_order: Dict[int, List[Optional[_Action]]]):
```

Comment:
Output looks like this (for num_local_stages=4 num_microbatches=8 group_size=8):

https://gist.github.com/H-Huang/2de7531dd6605b7a7eee6b95295291c9